misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/CoverageReport.java

```diff
@@ -123,13 +138,34 @@ public ModuleCoverage generateReport(JarResolver jarResolver) throws IOException
     }
 
     private IBundleCoverage analyzeStructure() throws IOException {
+
         final CoverageBuilder coverageBuilder = new CoverageBuilder();
         final Analyzer analyzer = new Analyzer(execFileLoader.getExecutionDataStore(), coverageBuilder);
         analyzer.analyzeAll(classesDirectory.toFile());
         return coverageBuilder.getBundle(title);
     }
 
+    private void createXMLReport(IBundleCoverage bundleCoverage) {
+
+        try {
+            XMLFormatter xmlFormatter = new XMLFormatter();
+            File reportFile = new File(target.getReportPath().resolve(
+                    this.module.moduleName().toString()).resolve("report.xml").toString());
+            reportFile.getParentFile().mkdirs();
+            IReportVisitor visitor = xmlFormatter.createVisitor(new FileOutputStream(reportFile));
+            visitor.visitInfo(execFileLoader.getSessionInfoStore().getInfos(),
+                    execFileLoader.getExecutionDataStore().getContents());
+
+            visitor.visitBundle(bundleCoverage,
+                    new DirectorySourceFileLocator(null, "utf-8", 4));
+
+            visitor.visitEnd();
+        } catch (IOException e) {
```

Comment:
If `generateReport()` throws an IOException, then shall we just have `createXMLReport` throw the IOException as well instead of catching it? 

@azinneera WDYT?

Reply:
yes we should throw the IOException